src/draw/lv_draw_private.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file lv_draw_private.h
+ *
+ */
+
+#ifndef LV_DRAW_PRIVATE_H
+#define LV_DRAW_PRIVATE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_draw.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+struct lv_draw_task_t {
+    lv_draw_task_t * next;
+
+    lv_draw_task_type_t type;
+
+    /**
+     * The area where to draw
+     */
+    lv_area_t area;
+
+    /**
+     * The real draw area. E.g. for shadow, outline, or transformed images it's different from `area`
+     */
+    lv_area_t _real_area;
+
+    /** The original area which is updated*/
+    lv_area_t clip_area_original;
+
+    /**
+     * The clip area of the layer is saved here when the draw task is created.
+     * As the clip area of the layer can be changed as new draw tasks are added its current value needs to be saved.
+     * Therefore during drawing the layer's clip area shouldn't be used as it might be already changed for other draw tasks.
+     */
+    lv_area_t clip_area;
+
+    volatile int state;              /*int instead of lv_draw_task_state_t to be sure its atomic*/
+
+    void * draw_dsc;
+
+    /**
+     * The ID of the draw_unit which should take this task
+     */
+    uint8_t preferred_draw_unit_id;
+
+    /**
+     * Set to which extent `preferred_draw_unit_id` is good at this task.
+     * 80: means 20% better (faster) than software rendering
+     * 100: the default value
+     * 110: means 10% worse (slower) than software rendering
+     */
+    uint8_t preference_score;
+
+};
+
+struct lv_draw_mask_t {
+    void * user_data;
+};
+
+struct lv_draw_unit_t {
```

Comment:
same as 2 reviews back

Reply:
Not needed to created new render units in the binding.